# This file is part of HiGFXback

# requires
REQUIRES="make-build opengl-registry-build"

PKG_CONFIG_PATH=/drm/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix=/drm

# build
make

# install
install -d $DESTDIR/drm/lib
install libEGL.so.1 $DESTDIR/drm/lib/libstubEGL.so.1
ln -sf libstubEGL.so.1 $DESTDIR/drm/lib/libEGL.so
ln -sf libstubEGL.so.1 $DESTDIR/drm/lib/libEGL.so.1
install libGL.so.1 $DESTDIR/drm/lib/libstubGL.so.1
ln -sf libstubGL.so.1 $DESTDIR/drm/lib/libGL.so
ln -sf libstubGL.so.1 $DESTDIR/drm/lib/libGL.so.1
install libGLESv1_CM.so.1 $DESTDIR/drm/lib/libstubGLESv1_CM.so.1
ln -sf libstubGLESv1_CM.so.1 $DESTDIR/drm/lib/libGLESv1_CM.so
ln -sf libstubGLESv1_CM.so.1 $DESTDIR/drm/lib/libGLESv1_CM.so.1
install libGLESv2.so.2 $DESTDIR/drm/lib/libstubGLESv2.so.2
ln -sf libstubGLESv2.so.2 $DESTDIR/drm/lib/libGLESv2.so
ln -sf libstubGLESv2.so.2 $DESTDIR/drm/lib/libGLESv2.so.2
install -d $DESTDIR/drm/lib/pkgconfig
install -m 644 egl.pc $DESTDIR/drm/lib/pkgconfig
install -m 644 gl.pc $DESTDIR/drm/lib/pkgconfig
install -m 644 glesv1_cm.pc $DESTDIR/drm/lib/pkgconfig
install -m 644 glesv2.pc $DESTDIR/drm/lib/pkgconfig

# build.pc
install -d $DESTDIR/drm/share/pkgconfig
cat > $DESTDIR/drm/share/pkgconfig/egl-opengl-stubs-build.pc << EOF
Name: EGL-OpenGL-Stubs
Version: 0.8
Description: EGL and OpenGL stubs for linking at build time
Requires: $REQUIRES

devel=\\
/drm/lib/libEGL.so \\
/drm/lib/libGL.so \\
/drm/lib/libGLESv1_CM.so \\
/drm/lib/libGLESv2.so \\
/drm/lib/pkgconfig/egl.pc \\
/drm/lib/pkgconfig/gl.pc \\
/drm/lib/pkgconfig/glesv1_cm.pc \\
/drm/lib/pkgconfig/glesv2.pc

exec=\\
/drm/lib/libEGL.so.1 \\
/drm/lib/libGL.so.1 \\
/drm/lib/libGLESv1_CM.so.1 \\
/drm/lib/libGLESv2.so.2
/drm/lib/libstubEGL.so.1 \\
/drm/lib/libstubGL.so.1 \\
/drm/lib/libstubGLESv1_CM.so.1 \\
/drm/lib/libstubGLESv2.so.2
EOF

exit

# patch
--- EGL-OpenGL-Stubs-0.8.orig/Configure
+++ EGL-OpenGL-Stubs-0.8/Configure
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) PREFIX=${1#*=};;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX/s|=.*|= $PREFIX|" Makefile
+
+sed -i "/^prefix/s|=.*|=$PREFIX|" egl.pc
+sed -i "/^prefix/s|=.*|=$PREFIX|" gl.pc
+sed -i "/^prefix/s|=.*|=$PREFIX|" glesv1_cm.pc
+sed -i "/^prefix/s|=.*|=$PREFIX|" glesv2.pc

# source
https://github.com/caramelli/EGL-OpenGL-Stubs/archive/v0.8/EGL-OpenGL-Stubs-0.8.tar.gz
